src/components/forms/Form.ts

```diff
@@ -19,41 +20,39 @@ export class Form extends Block {
   };
   constructor(props: IncomingProps) {
     super(props);
-    this.setProps({
-      onSubmit: (event: Event) => {
-        event.preventDefault();
-        const form = document.forms.namedItem(props.name);
-        if (typeof form !== 'object' || !(form instanceof HTMLFormElement)) {
-          return false;
-        }
-        const formData: FormData = new FormData(form);
-        const data: Record<string, unknown> = {};
-        for (const [key, value] of formData.entries()) {
-          data[key] = value;
-        }
-        const state: EventState = {errorMsgs: {}};
+    this.props.onSubmit = (event: Event) => {
+      event.preventDefault();
+      if (Store.state.isLoading) return false;
+      const form = document.forms.namedItem(props.name);
+      if (typeof form !== 'object' || !(form instanceof HTMLFormElement)) {
+        return false;
+      }
+      const formData: FormData = new FormData(form);
+      const data: Record<string, unknown> = {};
+      for (const [key, value] of formData.entries()) {
+        data[key] = value;
+      }
+      const state: EventState = {errorMsgs: {}};
+      this.listDescendants((block: Block) => {
+        block.emit('submit', event, state); // для валидации инпутов формы
+      });
+      if (Object.keys(state.errorMsgs).length === 0) {
```

Comment:
Можно упростить условие
```suggestion
      if (!Object.keys(state.errorMsgs).length) {
```